src/App.tsx

```diff
@@ -1,35 +1,62 @@
+/* eslint-disable no-console */
 import React from 'react';
 import './App.scss';
 
+import { Clock } from './Clock';
+
 function getRandomName(): string {
   const value = Math.random().toString().slice(2, 6);
 
   return `Clock-${value}`;
 }
 
-const App: React.FC = () => {
-  const date = new Date();
-  const clockName = getRandomName();
-
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    // ...
-  }, 1000);
-
-  // this code stops the timer
-  clearInterval(timerId);
-
-  return (
-    <div className="App">
-      <h1>React clock</h1>
+type State = {
+  hasClock: boolean;
+  clockName: string;
+};
 
-      <div className="clock">
-        <strong>{clockName}</strong>
-        {' time is '}
-        {date.toLocaleTimeString()}
+class App extends React.Component<{}, State> {
+  state = {
+    hasClock: true,
+    clockName: getRandomName(),
+  };
+
+  timerId = 0;
+
+  componentDidMount() {
+    document.addEventListener('contextmenu', () => {
+      this.setState({ hasClock: false });
+    });
+
+    document.addEventListener('click', () => {
+      this.setState({ hasClock: true });
+    });
+
+    this.timerId = window.setInterval(() => {
+      this.setState({ clockName: getRandomName() });
+    }, 3300);
+  }
+
+  componentDidUpdate(_: any, prevState: State) {
+    console.log(`Renamed from ${prevState.clockName} to ${this.state.clockName}`);
+  }
```

Comment:
This logic should be moved to `Clock`.  When we hide clock you don't need to show these logs in console:
<img width="1440" alt="image" src="https://user-images.githubusercontent.com/52370890/178545481-193d525c-7b31-4189-a27f-2233bd22e5a0.png">